cmd/kubectl/kubectlcobra/printer_adapter.go

```diff
@@ -0,0 +1,44 @@
+package kubectlcobra
+
+import (
+	"io"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+type KubectlPrinterAdapter struct {
```

Comment:
Looks like this sends objects to a channel to print them -- is that correct?

Reply:
I have added some documentation to this file to clarify. But the KubectlPrinterAdapter creates a `ToPrinter` function that instead of printing, turns the information into events and publishes them to the channel. This was the only way I could find to capture progress from the `ApplyOptions` without making changes to upstream.